src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +392,48 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    getThumbnailUrl(template: string, representation: string, tokenString: string): string {
+        const thumbnailBaseUrl = template.replace('{+asset_path}', representation === 'jpg' ? '' : '1.png');
+        return `${thumbnailBaseUrl}?access_token=${tokenString}`;
+    }
+
+    assignItemThumbnailUrl(item: BoxItem, index: number, template: string, representation: string): void {
+        const { token } = this.props;
+
+        TokenService.getToken(getTypedFileId(item.id), token).then(tokenString => {
+            if (tokenString) {
+                // Asset path for thumbnail is simply empty string.
+                const newCollection = { ...this.state.currentCollection };
+                newCollection.items = newCollection.items.map((newItem, newIndex) => {
+                    if (newIndex === index) {
+                        const clonedItem = cloneDeep(item);
+                        clonedItem.thumbnailUrl = this.getThumbnailUrl(template, representation, tokenString);
+                        return clonedItem;
+                    }
+                    return newItem;
+                });
+                this.setState({ currentCollection: newCollection });
+            }
+        });
+    }
+
+    assignCollectionThumbnailUrls(): void {
+        const { currentCollection } = this.state;
+        if (currentCollection.items) {
```

Comment:
Instead of iterating over the collection.items and for each item creating a newCollection and setting that to the state, could we instead `.map` the currentCollection to add the thumbnailUrl and then finally set to state in `assignCollectionThumbnailUrls`?